lib/rules/no-hooks-from-ancestor-modules.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * @fileoverview forbid the use of hooks from ancestor modules
+ * @author Raymond Cohen
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const utils = require("../utils");
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+const NESTABLE_HOOK_NAMES = ["afterEach", "beforeEach"];
```

Comment:
Do we also need to cover `setup` and `teardown` in this rule?

Reply:
Were `hooks.setup()` and `hooks.teardown()` supported at some point?

Reply:
Great question, I don't actually know offhand! I've raised the question in https://gitter.im/qunitjs/qunit.

Reply:
In chat we determined that `hooks.setup` and `hooks.teardown` were never valid